### PR TITLE
Apply strip_tags to attributes in loop template

### LINKED ISSUE #19491: HTML not escaped in product title
Hi,

even if it's a bad practice, it happens that editors/store managers use **HTML tags inside titles** –
e.g. a `<br>` to break in more lines a long title or a `<small></small>` to style a more or less relevant part.

This can cause **output issues** every time the title is used in contexts other than default HTML title (e.g. in link title attributes, data-attributes, …), where HTML tags shouldn't be placed.

WC 3+ uses `$product->get_name()` – which has no hook to escape HTML.
However, in most cases, **HTML is escaped** with `sanitize_title( $product->get_name() )` or `esc_html ( $product->get_name () )`.
But in some cases, e.g.  various`add_to_cart_description()`, **HTML is not escaped**, and it may break the output.

There may be other cases where it is not escaped, you may check if, every time `$product->get_name()` is used, it's better to escape HTML or leave it to output in DOM, whenever
editors put HTML tags in the title.

Thanks

### PR DESCRIPTION
**Pull request by peterfabian**
Fixes #19491 

Testing instructions:
1. Create a product with html tags in the title, e.g. <em>Very</em> nice beanie.
2. Go to /shop/ page (tested with Storefront theme).
3. Look for `aria-label` attribute for `/shop/?add-to-cart=X` anchor; before the fix, you should see `aria-label="Add “<em>Very</em> nice Beanie” to your cart"`.
4. After the fix, the `<em>` tags are stripped: `aria-label="Add “Very nice Beanie” to your cart"`

### CODE REVIEW

**Review by mikejolley — CHANGES_REQUESTED** [review]
The problem with tackling this in the template is:

1. The template version needs a bump to indicate the change (from 3.3 to 3.4)
2. This affects all attributes. I'm not certain but there could be instances where tags are wanted in the attribute e.g. for tooltips?

I'd recommend tackling this in the `woocommerce_template_loop_add_to_cart` function and just stripping tags for the aria-label attribute instead. Thats done outside of template and fixes this problem only.

**Comment by peterfabian** [comment]
Thanks for the review. I updated the PR according to the suggestion.

File: includes/wc-template-functions.php [hunk 1]
```diff
@@ -893,6 +893,10 @@ function woocommerce_template_loop_add_to_cart( $args = array() ) {
 
 			$args = apply_filters( 'woocommerce_loop_add_to_cart_args', wp_parse_args( $args, $defaults ), $product );
 
+			if ( array_key_exists( 'attributes', $args ) && array_key_exists( 'aria-label', $args['attributes'] ) ) {
```
Comment (mikejolley) [inline includes/wc-template-functions.php]: Any reason using this instead of `isset`? These could both be combined into a single isset() statement :)

**Comment by claudiosanches** [comment]
Why not include into `wc_implode_html_attributes()` since already is applying `esc_attr()`, there is the place where we should avoid include HTML.
Not to mention that there avoid us to update a template with conditional code.

**Comment by peterfabian** [comment]
@claudiosanches as Mike mentioned, that would affect all attributes, then, which might be not desired? 
Plus it would affect all select input boxes created by `woocommerce_wp_select` from `wc-meta-box-functions.php`. 

Given that tags are not forbidden in quoted attributes, I'm not sure if they should always be just stripped. 

If you (@claudiosanches @mikejolley) think it's better handled elsewhere, please let me know.

**Review by mikejolley — APPROVED** [review]
👍 This is what I had in mind.

@claudiosanches Some elements like data attributes/tooltips may want tags so it's best to target specific attributes IMO since `esc_attr` doesn't help us.

**Comment by claudiosanches** [comment]
Ohh sorry, I just show now that we are not changing a template, so no problem.

**Pull request merged**